device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    /**
+     * Opens this multiplexing client. At least one device client must be registered prior to calling this.
+     * <p>
+     * This call behaves synchronously, so if it returns without throwing, then all registered device clients were
+     * successfully opened.
+     * <p>
+     * If this client is already open, then this method will do nothing.
+     * <p>
+     * @throws IOException If any IO errors occur while opening the multiplexed connection.
+     */
+    public void open() throws IOException
+    {
+        if (deviceClientList.size() < 1)
+        {
+            throw new UnsupportedOperationException("Must register at least one device client before opening a multiplexed connection");
+        }
+
+        log.info("Opening multiplexing client");
+        this.deviceIO.open();
+        log.info("Successfully opened multiplexing client");
+    }
+
+    /**
+     * Close this multiplexing client. This will close all active device sessions as well as the AMQP connection.
+     * <p>
+     * If this client is already closed, then this method will do nothing.
+     * <p>
+     * Once closed, this client can be re-opened. It will preserve the previously registered device clients.
+     * <p>
+     * @throws IOException If any exception occurs while closing the connection.
+     */
+    public void close() throws IOException
+    {
+        log.info("Closing multiplexing client");
+        for (DeviceClient deviceClient : this.deviceClientList)
+        {
+            deviceClient.closeFileUpload();
+        }
+
+        if (this.deviceIO != null)
+        {
+            this.deviceIO.multiplexClose();
+        }
+
+        log.info("Successfully closed multiplexing client");
+    }
+
+    /**
+     * Add a device to this multiplexing client. This method may be called before or after opening the multiplexed
+     * connection, but will behave differently depending on when it was called.
+     * <p>
+     * Up to 1000 devices can be registered on a multiplexed AMQPS connection, and up to 500 devices can be registered on a
+     * multiplexed AMQPS_WS connection.
+     * <p>
+     * If the multiplexing client is already open, then this device client will automatically
+     * be opened, too. If the multiplexing client is not open yet, then this device client will not be opened until
+     * {@link MultiplexingClient#open()} is called.
+     * <p>
+     * If the multiplexed connection is already open, then this is an asynchronous operation. You can track the state of your
+     * device session using the {@link DeviceClient#registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback, Object)}.
+     * That callback will execute once your device session has successfully been added to the existing multiplexed connection and
+     * is ready to send telemetry.
+     * <p>
+     * Any proxy settings set to the provided device client will be overwritten by the proxy settings of this multiplexing client.
```

Comment:
I could add something like that, but I think the typical use case will involve customers setting the proxy settings at the MultiplexingClient level, and not also setting them at each device client level. Because of that, if I add a warning log here, customers will hit it even when using our library correctly.

```java
MultiplexingClient multiplexingClient = new MultiplexingClient(protocol, someProxySettings);
DeviceClient[] multiplexedDeviceClients = new DeviceClient[MULTIPLEXED_DEVICE_CLIENT_COUNT];

for (int i = 0; i < MULTIPLEXED_DEVICE_CLIENT_COUNT; i++)
{
    //Users would have to do this to avoid the warning log if we add it, which feels unnecessary
    multiplexedDeviceClients.setProxySettings(someProxySettings);
}

```